src/PhpSpreadsheet/Writer/Xlsx/Worksheet.php

```diff
@@ -128,6 +144,213 @@ public function writeWorksheet(PhpspreadsheetWorksheet $worksheet, $stringTable
+        $objWriter->writeAttribute('xmlns:r', 'http://schemas.openxmlformats.org/officeDocument/2006/relationships');
+
+        $objWriter->writeAttribute('xmlns:xdr', 'http://schemas.openxmlformats.org/drawingml/2006/spreadsheetDrawing');
+        $objWriter->writeAttribute('xmlns:x14', 'http://schemas.microsoft.com/office/spreadsheetml/2009/9/main');
+        $objWriter->writeAttribute('xmlns:xm', 'http://schemas.microsoft.com/office/excel/2006/main');
+        $objWriter->writeAttribute('xmlns:mc', 'http://schemas.openxmlformats.org/markup-compatibility/2006');
+        $objWriter->writeAttribute('mc:Ignorable', 'x14ac');
+        $objWriter->writeAttribute('xmlns:x14ac', 'http://schemas.microsoft.com/office/spreadsheetml/2009/9/ac');
+
+        // sheetPr
+        $this->writeSheetPr($objWriter, $pSheet);
+
+        // Dimension
+        $this->writeDimension($objWriter, $pSheet);
+
+        // sheetViews
+        $this->writeSheetViews($objWriter, $pSheet);
+
+        // sheetFormatPr
+        $this->writeSheetFormatPr($objWriter, $pSheet);
+
+        // cols
+        $this->writeCols($objWriter, $pSheet);
+
+        // sheetData
+        $objWriter->startElement('sheetData');
+
+        $objWriter->flush();
+
+        return $this;
+    }
+
+    /**
+     * @param PhpspreadsheetWorksheet $pSheet
+     * @param array $pStringTable
+     * @return Worksheet
+     * @throws \PhpOffice\PhpSpreadsheet\Exception
+     */
+    public function writeSheetDataPortion(PhpspreadsheetWorksheet $pSheet, $pStringTable = [])
+    {
+        // Flipped stringtable, for faster index searching
+        $aFlippedStringTable = $this->getParentWriter()->getWriterPartStringTable()->flipStringTable($pStringTable);
+        $objWriter = $this->objWriters[$this->getParentWriter()->getSpreadsheet()->getActiveSheetIndex()];
+        // Get column count
+        $colCount = Coordinate::columnIndexFromString($pSheet->getHighestColumn());
+        $colCount = $this->maxColCount = max($colCount, $this->maxColCount);
+
+        // Highest row number
+        $highestRow = $pSheet->getHighestRow();
+
+        // Loop through cells
+        $cellsByRow = [];
+
+        foreach ($pSheet->getCoordinates() as $coordinate) {
+            $cellAddress = Coordinate::coordinateFromString($coordinate);
+            $cellsByRow[$cellAddress[1]][] = $coordinate;
+        }
+        //avoid to foreach not existed rows
+        $currentRow = $this->lastHighestRow;
+        while ($currentRow++ < $highestRow) {
+            $stringCurrentRow = (string)$currentRow;
+            // Get row dimension
+            $rowDimension = $pSheet->getRowDimension($currentRow);
+            // Write current row?
+            $writeCurrentRow = isset($cellsByRow[$stringCurrentRow]) || $rowDimension->getRowHeight() >= 0 || $rowDimension->getVisible() == false || $rowDimension->getCollapsed() == true || $rowDimension->getOutlineLevel() > 0 || $rowDimension->getXfIndex() !== null;
```

Comment:
this check failure of phpstan is a bug , please view this: https://github.com/phpstan/phpstan/issues/3170